CMakeLists.txt

```diff
@@ -1,155 +1,143 @@
 ### CMakeList for MC Hydra
 
-cmake_minimum_required(VERSION 3.0)
-cmake_policy(SET CMP0054 OLD)
-
-# project name
-project(Hydra)
-
-# The version number.
-set (Hydra_VERSION_MAJOR 1)
-set (Hydra_VERSION_MINOR 0)
-set (Hydra_VERSION_PATCH 0)
-
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
- MESSAGE(STATUS "===> Compiling with Clang")
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
- MESSAGE(STATUS "===> Compiling with GCC")
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
- MESSAGE(STATUS "===> Compiling with Intel")
-elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
- MESSAGE(STATUS "===> Compiling with MSVC")
-endif()
+cmake_minimum_required(VERSION 3.1)
+
+# Project name and version
+project(Hydra LANGUAGES CXX
+              VERSION 1.0.0)
 
-# warn user if system is not UNIX
+# Warn user if system is not UNIX
 if(NOT UNIX)
-  message(FATAL_ERROR "This is an unsupported system.")
```

Comment:
Warn about a system not being supported, never kill it. Someone may add support for a new system down the road! (And, even the comment says it will be a warning)

Reply:
This is a matter of opinion. The person implementing support for another system can perfectly change it. I wll think about....

Reply:
I know it's a matter of opinion, but you said "Warn the user" so I set to to warn. Please tell me if you want it to be a fatal error, and I'll change it, so you can accept the PR (you can also push to this branch if you want changes).

Reply:
I would like to keep this like that for the moment. Please, can you change it?

Reply:
Augusto, why do you want to make life complicated to the random user who would like to try it on some other architecture? That's it's problem in the end, no? Why would you even care and make it impossible with FATAL?

Reply:
Hi Eduardo, Hydra is not supported for other platforms than linux.  I make a huge deal of effort to keep the code as agnostic as possible, but more testing is necessary before to add support for other systems. 

Build is just required to compile and run the examples. I tested the examples in linuxes machines, I don't know how they would behave in other systems. User who hacked it and ran on other systems are encouraged to report their experience in the hydra-library-users google group. 

Henry, please keep Fatal_error.

Reply:
I am not saying above that you should give support for other platforms. I am saying you should not close the door and make it impossible or really hard for anyone to even try the package in a platform you have not tested. That's all. Having to hack for even trying to build is not great IMHO. Your call.

Reply:
"... really hard for anyone to even try the package in a platform you have not tested."
This comment is misleading Eduardo, building is just necessary for the examples ...
It is a header only library.  Nothing prevent users from other platforms of include the code in their source trees.

Reply:
I rest my case. Personally, the first thing I try with a new package is to build it and/or run some examples. If I cannot get it to work on my system that's frustrating from the onset. Maybe that's just me. Fine.

Reply:
Why would someone in approach  not try to hack the package? You left out that in both cases you may have to hack Hydra to get it to work, the approach 1 just means you hack 1 less line. (not everyone is great at reading CMake, by the way). Not immediately killing it in a FATAL_ERROR would help them see what needs to be hacked immediately on their system and help them start fixing things.

Remember, this will *only* make a difference for someone brave enough to try it on a different system (FreeBSD, Windows, etc.). And in either approach, they *still get the unsupported warning*.

The only benefit I see to Approach 1 is if the only required change is the one single line, then you will learn that it really does work on that system, vs. a small chance that someone in Approach 2 won't bother to report that the warning is not needed on their system if it just works.

I'm 100% okay with either approach, though I do like Approach 2 a bit better. I *think* the result of this thread was to keep FATAL_ERROR, so I'll do that and let Augusto change it if he decides to go with Approach 2.

Reply:
I muddled approaches 1 and 2, corrected above.